app/controllers/users.controller.js

```diff
@@ -1,41 +1,77 @@
+var jwt = require('jsonwebtoken');
+var config = require('../../config/config');
+var verification = require('../utils/verification.util');
 var User = require('mongoose').model('User');
 
-exports.create = function(req, res, next) {
-    var user = new User(req.body);
-    user.save(function(err) {
-        if (err) {
-            return next(err);
-        } else {
-            res.json(user);
-        }
-    });
-};
+const EMAIL_REGEX = /^.+@gatech.edu$/i;
+const MIN_PASSWORD_LENGTH = 6;
+const TOKEN_EXPIRATION_TIME = '7 days';
 
-exports.list = function(req, res, next) {
-    User.find({}, function(err, users) {
-        if (err) {
-            return next(err);
-        } else {
-            res.json(users);
-        }
-    });
-};
+exports.createAccount = function(req, res) {
+    var email = (req.body.email) ? req.body.email.trim() : '';
+    var password = (req.body.password) ? req.body.password.trim() : '';
+    var firstName = (req.body.firstName) ? req.body.firstName.trim() : '';
+    var lastName = (req.body.lastName) ? req.body.lastName.trim() : '';
+
+    if (!EMAIL_REGEX.test(email)) {
+        res.json({successful: false, text: 'Please provide a valid Georgia Tech email address'});
+
+    } else if (password.length < MIN_PASSWORD_LENGTH) {
+        res.json({successful: false, text: 'Please provide a password that is at least 6 characters long.'});
+
+    } else if (firstName === '') {
+        res.json({successful: false, text: 'Please provide a first name.'});
+
+    } else if (lastName === '') {
+        res.json({successful: false, text: 'Please provide a last name.'});
 
-exports.read = function(req, res) {
-    res.json(req.user);
+    } else {
+        var user = new User({
+            email: email,
+            password: password,
+            firstName: firstName,
+            lastName: lastName
+        });
+        // save the user
+        user.save(function(err) {
+            if (err) {
+                return res.json({successful: false, text: 'The email address, ' +  user.email
+                + ' is already associated with another account.'});
+            }
+            res.json({successful: true, text: 'A new account was created for ' + user.email + '!'});
+        });
+    }
 };
 
-exports.userByID = function(req, res, next, id) {
-    User.findOne({
-            _id: id
-        },
-        function(err, user) {
+exports.login = function(req, res) {
+    var failureMsg = 'The email or password you provided was incorrect.';
+
+    User.findOne({email: req.body.email})
+        .select('+password')
+        .exec(function(err, user) {
             if (err) {
-                return next(err);
+                throw err;
+            } else if (!user) {
+                res.status(401).send({successful: false, text: failureMsg});
             } else {
-                req.user = user;
-                next();
+                // check if password matches
```

Comment:
I'm thinking we don't need the comments in this section. Everything seems pretty self-explanatory. I feel like the comments are more distracting than actually helping to clarify.

Reply:
OK. I will remove them if they are not helpful.